README.md

```diff
@@ -1,264 +1,338 @@
 
 # AWS IoT Device SDK for Embedded C
 
+## Table of Contents
+
+* [Overview](#overview)
+    * [License](#license)
+    * [Features](#features)
+        * [coreMQTT](#coremqtt)
+        * [coreHTTP](#corehttp)
+        * [coreJSON](#corejson)
+        * [AWS IoT Device Shadow](#aws-iot-device-shadow)
+    * [AWS Collection of Metrics](#aws-collection-of-metrics)
+* [Versioning](#versioning)
+* [Releases](#releases)
+    * [202011.00](#20201100)
```

Comment:
This list is going to get really long lol

Reply:
Well, that's one way to write a novel

Reply:
How bout collapsible markdown?
https://gist.github.com/joyrexus/16041f2426450e73f5df9391f7f7ae5f

Reply:
Is there a way to autogenerate a toc from markdown?

Reply:
There's scripts that can generate it given a markdown file, but no markdown syntax for a ToC.